### DISCUSSION TITLE
Can we dynamically modify Artillery.io think time at runtime?

### DISCUSSION THREAD

**Discussion by nullsocket** [artilleryio/artillery#1976] [Artillery]
Is there any way to dynamically modify Artillery.io think time at runtime?

Basically what I'm looking into is if I can change the think value based off of data grabbed from the response to either slow, speed up or essentially briefly pause the requests being sent for vusers preferably by using beforeRequest and afterRequest processors

**Comment by hassy** [comment]
Hi @nullsocket 👋 You can set thinktimes dynamically with variables, e.g.:

```yaml
- think: "{{ timeInSec }}"
```

And `timeInSec` can be set dynamically by an `afterResponse` hook function on the previous request:

```yaml
- get:
    url: "/todos"
    afterResponse: "setThinkTime"
```

```js
// Set a larger think time if the request takes longer than 500ms
function setThinkTime(req, res, userContext, events, next) {
  if (res.timings?.phases?.total) {
    if (res.timings.phases.total > 500) {
      userContext.vars['timeInSec'] = 3;
    } else {
      userContext.vars['timeInSec'] = 1;
    }
  }
  
  return next();
}
```

More docs here: https://www.artillery.io/docs/reference/engines/http#afterresponse-hooks

**Reply by nullsocket** [reply]
Thanks appreciate the detailed response
